### ISSUE TITLE
Listener that listens to false usage, no permission, ...

### ISSUE THREAD

**Issue by Paul2708** [Paul2708/simple-commands#12]
## Description
Possibility to listen to certain (custom) events.

**Possible events:**
- player tries to execute a command with missing permission
- player types in a false usage
- command execution in general (with provided command annotation information)

## Usage
**Implementation idea**
Dispatch custom events by using the built-in event-system.
More details about the event system can be found [here](https://github.com/Paul2708/simple-commands/issues/12#issuecomment-544461547).

**Comment by lineargraph** [comment]
Spigot has their own event dispatchment system which we could use for this.
See: 

 - [Event base class](https://hub.spigotmc.org/javadocs/spigot/org/bukkit/event/Event.html)
     - [Example implementations](https://hub.spigotmc.org/stash/projects/SPIGOT/repos/bukkit/browse/src/main/java/org/bukkit/event)
 - [Dispatching events](https://hub.spigotmc.org/javadocs/spigot/org/bukkit/plugin/PluginManager.html#callEvent-org.bukkit.event.Event-)

**Comment by Paul2708** [comment]
Good point. Using the build-in event-system by Spigot makes a lot more sense too me.

**Comment by Paul2708** [comment]
A good entry point would be `CommandDelegator`. Most of the checks are performed in `#execute(...)`.
